packages/wdio-browserstack-service/README.md

```diff
@@ -141,8 +135,53 @@ services: [
 ]
 ```
 
+### preferScenarioName
+
+Cucumber only. Set the BrowserStack Automate session name to the Scenario name if only a single Scenario ran.
+Useful when running in parallel with [wdio-cucumber-parallel-execution](https://github.com/SimitTomar/wdio-cucumber-parallel-execution).
+
+Type: `Boolean`<br />
+Default: `false`
+
+### sessionNameFormat
+
+Customize the BrowserStack Automate session name format.
+
+Type: `Function`<br />
+Default (Cucumber/Jasmine): `(config, capabilities, suiteTitle) => suiteTitle`<br />
+Default (Mocha): `(config, capabilities, suiteTitle, testTitle) => suiteTitle + ' - ' + testTitle`
+
+### sessionNameOmitTestTitle
+
+Mocha only. Do not append the test title to the BrowserStack Automate session name.
+
+Type: `Boolean`<br />
+Default: `false`
+
+### sessionNamePrependTopLevelSuiteTitle
+
+Mocha only. Prepend the top level suite title to the BrowserStack Automate session name.
+
+Type: `Boolean`<br />
+Default: `false`
+
+### setSessionName
+
+Automatically set the BrowserStack Automate session name.
+
+Type: `Boolean`<br />
+Default: `true`
+
+### setSessionStatus
```

Comment:
Just curious, I would love to understand when people are interested to turn this off. Is it because these additional request slow down the test run?

Reply:
For my use case I'd like to set the session status/reason in an `after` hook without it being overridden.

Reply:
Why are you setting the name in the after hook? Just curious.

Reply:
It's what I would do as well. I would like to write the path of the module there as session name.
Unless browserstack starts supporting it's within a session.

Reply:
For session name:

We've seen instances of incompatibility of the cucumber wdio parallel plugin and the service's naming logic at a scenario level.

In this case, users have opted to not use the service entirely.

Another one was where describe blocks were written in Korean, and Browserstack currently doesn't support those characters in the name.

While the above are either bugs or fixable limitations, at least the flexibility helps users to fix things themselves, and not have to troubleshoot a behaviour where they set context manually and then it gets overridden on BrowserStack via the service.

For session status:

I've seen patterns where the session reason is set to a grouped message based on all the assertions that have failed in a session.

Something like
"1 of 3 tests failed"

Currently we set the reason based on a single failure

Reply:
Makes sense to me, thanks for clarifying @alliv8

Reply:
> Why are you setting the name in the after hook? Just curious.

Not the name but the status (passed/failed) and a descriptive `reason` that conveys additional information if `passed`.